### PR TITLE
feat: add API endpoint to consume tower webhook notifications

### PR DESCRIPTION
**Pull request by brhatiga**

### CODE REVIEW

File: src/openapi/openapi.yml [hunk 1]
```diff
@@ -351,16 +351,95 @@ components:
...
           type: boolean
         finished_at:
           type: string
           format: date-time
-          x-nullabe: true
+          x-nullable: true
         failed:
           type: boolean
 
+    TowerWebhookNotification:
```
Comment (fpob) [inline src/openapi/openapi.yml:364]: Is notification payload the same also for workflow jobs? Workflows don't have fields like project, playbook, and maybe few others.

Comment (brhatiga) [reply src/openapi/openapi.yml:364]: Added the following for job-workflows:
https://github.com/resource-hub-dev/rhub-api/pull/26/files#diff-3d6fdebe82daa0e03f81cf6f62f17d0c33268587a8914d89d7b4dffef2d83b99R446-R449

File: src/rhub/api/tower.py [hunk 2]
```diff
@@ -282,3 +282,55 @@ def get_job_stdout(job_id):
...
+                and password == current_app.config['WEBHOOK_PASS']):
+            return {'sub': 'webhook'}    # successful authentication
+
+    except KeyError as e:
+        logger.exception('Missing tower webhook notification credential(s)'
+                         f' {e}; notification ignored')
+
+    logger.warning('Incorrect tower webhook notification credentials supplied;'
+                   ' notification ignored')
+    return None    # failed authentication
```
Comment (fpob) [inline src/rhub/api/tower.py:303]: Maybe raise `Unauthorized` (from `werkzeug.exceptions`) there to be 100% sure?

File: src/rhub/api/tower.py [hunk 3]
```diff
@@ -282,3 +282,55 @@ def get_job_stdout(job_id):
...
+    #      - update_cluster_status(id)  [database operation]
+    #      - send_email(user.email, message) or submit the notification to Hydra?
+
+    # inspect json payload to ensure certain fields are present
+    try:
+        jobId = request.json['id']
+        jobStatus = request.json['status']
+    except Exception as e:
+        logger.exception(f'Unexpected tower webhook notification json; missing {e}')
+        return {'error, unexpected tower webhook notification json; '
```
Comment (fpob) [inline src/rhub/api/tower.py:327]: You should use `connexion.problem` to return problems as defined in the API spec. `problem(400, ..., ..., ext={'missing': ...})`

File: src/rhub/api/tower.py [hunk 4]
```diff
@@ -282,3 +283,60 @@ def get_job_stdout(job_id):
...
+        logger.exception('Missing tower webhook notification credential(s)'
+                         f' {e}; notification ignored')
+
+    logger.warning('Incorrect tower webhook notification credentials supplied;'
+                   ' notification ignored')
+
+    raise Unauthorized('Incorrect tower webhook notification credentials'
+                       ' supplied')
+
+    return None    # failed authentication
```
Comment (fpob) [inline src/rhub/api/tower.py:308]: dead code. `return` can never be executed because of `raise` above

**Review by fpob — CHANGES_REQUESTED** [review]

File: src/rhub/api/tower.py (hunk 4 above)
Comment (brhatiga) [reply src/rhub/api/tower.py]: thanks!  totally missed that!

**Review by fpob — APPROVED** [review]

**Review by thaorell — APPROVED** [review]
LGTM

**Pull request merged**